camayoc/tests/qpc/cli/test_credentials.py

```diff
@@ -710,3 +710,69 @@ def test_clear_all(isolated_filesystem, qpc_server_config, cleaning_data_provide
     )
     assert "No credentials exist yet." in output
     assert exitstatus == 0
+
+
+@pytest.mark.upgrade_only
+def test_edit_existing_credential_username(qpc_server_config, source_type):
+    """Check that credential that existed before upgrade can be edited.
+
+    We want to check all credential types, but the only field that is common
+    for all of them is the name - and this is one field that CLI can not
+    change. So we try to change username instead, but RHACS and OpenShift
+    token do not have it.
+
+    :id: 93dd6711-ef58-4370-9e4a-d427785a9bf9
+    :description: Edit existing credential
+    :steps:
+        1) Select a credential of specified type
+        2) Edit credential, changing the username
+        3) Verify that credential was changed
+        4) Edit credential again, restoring old username
+    :expectedresults: Credential is modified and changes are saved.
+    """
+    new_username = utils.uuid4()
+
+    output, _ = pexpect.run(
+        "{} -v cred list --type {}".format(client_cmd, source_type),
+        encoding="utf8",
+        withexitstatus=True,
+    )
+    try:
+        all_credentials = json.loads(output)
+    except ValueError:
+        pytest.skip("There are no credentials of this type")
+
+    credential = random.choice(all_credentials)
+    credential_name = credential.get("name")
+    if not credential.get("username"):
+        pytest.skip("Credential does not store username")
```

Comment:
I'm probably missing context here, but I'm under the impression we might want to only skip if no credentials have `username`. I suggest we randomly select from credentials pre-filtered, with the filter being "credential has `username` key".

Reply:
You are right. Fixed in 4944aea826887cd7b2ae4420083e16bca0a592fa